docs/modules/ROOT/pages/explanations/decisions/api-design.adoc

```diff
@@ -0,0 +1,250 @@
+= Decision on API Design
+
+== Problem
+
+This decision is about how we can achieve a common look as defined https://kb.vshn.ch/app-catalog/explanations/why_exists.html#_standard_way_to_provide_services[here].
+
+The ideal way would be, that we can re-use the exact same XRD for each set of services.
+For example all implementations for PostgreSQL have exactly the same API for the user.
+Due to the nature of the product, not all implementations provide the same functionality.
+The set of available features is most likely not the same between most providers.
+Also with Crossplane we're a bit limited on what's possible with translating the source API spec into the target API spec:
+
+* No conditionals in compositions
+* Only a limited set of transformations
+* Can't create random values
+
+[NOTE]
+This is not a complete list of limitations.
+
+== Requirements
+
+* APIs for a set of similar services feel similar
+* Only minimal configuration is needed for a default case
+* Unique features of a service provider can be leveraged
+
+== Proposals
+
+=== Proposal 1: Use same XRD with provider specific fields
+
```

Comment:
I think over time this looks more like an anti-pattern of Kubernetes.
Consider adding AWS and GCP, eventually you'll end up with maximum 3 common parameters, the rest is very provider-specific and in `.spec.parameters.{aws,gcp,exoscale,vshn}` keys. Might as well create different CRDs for each provider and accept that those 3 common parameters are duplicated.

Reply:
I've added the remark as a disadvantage.